nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java

```diff
@@ -156,6 +163,59 @@ private void processClassPath(String[] args) {
     return Sets.union(nonNullPositions, jiNonNullParams).immutableCopy();
   }
 
+  @Override
+  public NullnessHint onDataflowVisitMethodInvocation(
+      MethodInvocationNode node,
+      Types types,
+      AccessPathNullnessPropagation.SubNodeValues inputs,
+      AccessPathNullnessPropagation.Updates thenUpdates,
+      AccessPathNullnessPropagation.Updates elseUpdates,
+      AccessPathNullnessPropagation.Updates bothUpdates) {
+    if (!config.isJarInferUseReturnAnnotations()) return NullnessHint.UNKNOWN;
+    Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(node.getTree());
+    Preconditions.checkNotNull(methodSymbol);
+    Symbol.ClassSymbol classSymbol = methodSymbol.enclClass();
+    String className = classSymbol.getQualifiedName().toString();
+    if (!lookupAndBuildCache(classSymbol)) return NullnessHint.UNKNOWN;
+    String methodSign = getMethodSignature(methodSymbol);
+    Map<Integer, Set<String>> methodArgAnnotations = lookupMethodInCache(className, methodSign);
+    if (methodArgAnnotations == null) return NullnessHint.UNKNOWN;
+    Set<String> methodAnnotations = methodArgAnnotations.get(RETURN);
+    if (methodAnnotations != null) {
+      if (methodAnnotations.contains("javax.annotation.Nullable")) {
+        LOG(DEBUG, "DEBUG", "Nullable return for method: " + methodSign);
+        return NullnessHint.HINT_NULLABLE;
+      }
+    }
+    return NullnessHint.UNKNOWN;
+  }
+
+  private static final boolean NULLABLE = true;
+
+  @Override
+  public boolean onOverrideMayBeNullExpr(
+      NullAway analysis, ExpressionTree expr, VisitorState state, boolean exprMayBeNull) {
+    if (!config.isJarInferUseReturnAnnotations()) return exprMayBeNull;
```

Comment:
A lot of the code here seems duplicated with the other method, please extract into a common method.